pmd-core/src/main/java/net/sourceforge/pmd/cpd/CPDConfiguration.java

```diff
@@ -170,7 +175,12 @@ public void postContruct() {
         }
         if (getRenderer() == null && getCPDRenderer() == null) {
             try {
-                setCPDRenderer(getCPDRendererFromString(getRendererName(), getEncoding()));
+                try {
+                    setRenderer(getCPDReportRendererFromString(getRendererName(), getEncoding()));
+                } catch (ClassCastException e) {
+                    // The renderer class configured is not using the new CPDReportRenderer interface...
+                    setCPDRenderer(getCPDRendererFromString(getRendererName(), getEncoding()));
+                }
```

Comment:
This is not necessary, since getCPDReportRendererFromString will call getCPDRendererFromString already and use the adapter to convert old CPDRenderers into new CPDReportRenderers.

Consequently this means, that when you implement a new CPDReportRenderer you have to implement the old interface as well, otherwise there is no way to use your renderer...

I think, I'll reimplement this logic here to use instanceof checks rather than classcastexceptions and try each interface in turn.